presto-hive/src/main/java/com/facebook/presto/hive/ForWarmStorageAwareFileOpener.java

```diff
@@ -0,0 +1,18 @@
+package com.facebook.presto.hive;
+
+import javax.inject.Qualifier;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target({FIELD, PARAMETER, METHOD})
+@Qualifier
+public @interface ForWarmStorageAwareFileOpener
```

Comment:
Let's rename this to `ForCachingFileOpener` because this is used to annotate `FileOpener` in `CachingFileOpener`.

Reply:
@shixuan-fan We might need another annotation for the CachingFileOpener in future, hence we might want to reserve the name ForCachingFIleOpener for that annotation, Currently the CachingFileOpener binding is by default hence no annotation is used. let me know what you think?